src/common/text.c

```diff
@@ -814,13 +814,22 @@ text_convert_invalid (const gchar* text, gssize len, GIConv converter, const gch
 gchar *
 text_fixup_invalid_utf8 (const gchar* text, gssize len, gsize *len_out)
 {
+#if GLIB_CHECK_VERSION (2, 52, 0)
+	gchar *result = g_utf8_make_valid (text, len);
```

Comment:
Since we try to target an older glib you have to add `G_GNUC_BEGIN_IGNORE_DEPRECATIONS`/`G_GNUC_END_IGNORE_DEPRECATIONS` around this function to silence complaints.